app/code/Magento/Catalog/Controller/Product/Compare/Remove.php

```diff
@@ -31,7 +31,7 @@ public function execute()
                 $product = null;
             }
 
-            if ($product && $product->isSalable()) {
+            if ($product) {
```

Comment:
I am not sure if we should remove `$product->isSalable()`. In what case will someone compare the non-saleable product and is it correct to allow such comparisons?

Reply:
I see it is to remove the compared products, which might be the correct behavior even though the product is OOS. I will wait for someone else to review as well.